pkg/broker/ingress/handler_test.go

```diff
@@ -124,11 +136,20 @@ func TestHandler(t *testing.T) {
 	}
 
 	client := nethttp.Client{}
-	defer client.CloseIdleConnections()
+	t.Cleanup(client.CloseIdleConnections)
```

Comment:
Prow isn't using go 1.14 yet.

Reply:
Ah darn, I can no longer live without T.Cleanup. I will try bumping the go version: https://github.com/knative/test-infra/pull/1953

Reply:
It looks like the only place in this PR that we want `t.Cleanup` rather than `defer` is line 195, which could just return a function for the caller to defer for now.

Actually, I think we would prefer `defer` on lines 144 and 147, so things are shutdown more quickly.

Reply:
It would certainly be possible to thread through a cleanup function, but it would complicate the test functions a bit. I can do this if prow can't be switched to go 1.14 yet, but if it can this seems like a good time to make the switch. I believe the plan is for the rest of knative to switch to go 1.14  before the next release as well.

Reply:
I've gone ahead and removed use of T.Cleanup for now, but it's not pretty. Hopefully I'll be able to revert it shortly.

Reply:
Ooph, that is uglier than I thought it would be. `t.Cleanup` will be much nicer.